clients/pkg/promtail/targets/cloudflare/client.go

```diff
@@ -0,0 +1,37 @@
+package cloudflare
+
+import (
+	"context"
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+)
+
+// Client is a wrapper around the Cloudflare API that allow for testing and being zone/fields aware.
+type Client interface {
+	LogpullReceived(ctx context.Context, start, end time.Time) (cloudflare.LogpullReceivedIterator, error)
+}
+
+type wrappedClient struct {
+	client *cloudflare.API
+	zoneID string
+	fields []string
+}
+
+func (w *wrappedClient) LogpullReceived(ctx context.Context, start, end time.Time) (cloudflare.LogpullReceivedIterator, error) {
+	return w.client.LogpullReceived(ctx, w.zoneID, start, end, cloudflare.LogpullReceivedOption{
+		Fields: w.fields,
+	})
+}
+
+var getClient = func(apiKey, zoneID string, fields []string) (Client, error) {
```

Comment:
Why is this `var getClient = func` and not `func getClient`? Is there a special Golang reason?

Reply:
I'm overriding it for testing purpose.

Reply:
Ah, never seen that pattern. Is it common or does it need a comment?

Reply:
There's a comment at the top line 10.